multicluster/cmd/link.go

```diff
@@ -457,11 +477,20 @@ func buildServiceMirrorValues(opts *linkOptions) (*multicluster.Values, error) {
 		return nil, fmt.Errorf("--log-format must be one of: plain, json")
 	}
 
+	if opts.disableGateway {
+		if opts.selector == fmt.Sprintf("%s=%s", k8s.DefaultExportedServiceSelector, "true") {
+			opts.selector = ""
```

Comment:
does this do anything? the code path for rendering the Link ignores this opt when disableGateway is true, right?

Reply:
`opts.selector = ""` is a noop, but this check is a bit annoying:
* If gateway is disabled, then we can't have a selector
* By default, `opts.selector` is set to the default selector, we are okay with the default selector being set, we just don't want a user to specify a selector as an override
* If the selector is not empty, or the default selector, then it means it's an override, in which case error.

I've written this in a more concise way, I was still making sense of how all of this should look in yday's commit.